src/libraries/System.IO.FileSystem/tests/FileStream/FileStreamConformanceTests.cs

```diff
@@ -253,6 +253,11 @@ public class NamedPipeFileStreamConnectedConformanceTests : ConnectedStreamConfo
         protected override async Task<StreamPair> CreateConnectedStreamsAsync()
         {
             string name = FileSystemTest.GetNamedPipeServerStreamName();
+            // https://github.com/dotnet/runtime/issues/55689
+            if (PlatformDetection.IsAndroid)
+            {
+                name = name.Substring(0, (name.Length - 10));
+            }
```

Comment:
cc: @danmoseley 

I think we should address this differently.  The issue here isn't specific to Android: all platforms have a limit on how long a UDS path can be.  POSIX requires that it be at least 92, but for some platforms it's 108, for others it's 104, etc.  Our named pipe implementation is based on UDS and uses the path prefix:
```C#
Path.Combine(Path.GetTempPath(), "CoreFxPipe_") + pipeName
```
where pipeName in this test is Guid.NewGuid().ToString("N").

As such, every platform could hit an issue, and it's just because a) this particular project has a longer than usual name, and b) this machine/platform must have a particularly long tmp directory path that we're hitting it here.

I'd rather we change FileSystemTest.GetNamedPipeServerStreamName() to just return a shorter name.  Something like (I've not tested this):
```C#
public static string GetNamedPipeServerStreamName()
{
    if (PlatformDetection.IsInAppContainer)
    {
        return @"LOCAL\" + Guid.NewGuid().ToString("N");
    }

    if (PlatformDetection.IsWindows)
    {
        return Guid.NewGuid().ToString("N");
    }

    const int MinUdsPathLength = 104; // required min is 92, but every platform we currently target is at least 104
    const int MinAvailableForSufficientRandomness = 5; // we want enough randomness in the name to avoid conflicts between concurrent tests
    string prefix = Path.Combine(Path.GetTempPath(), "CoreFxPipe_");
    int availableLength = MinUdsPathLength  - prefix.Length;
    Assert.True(availableLength >= MinAvailableForSufficientRandomness, int.MaxValue, $"UDS prefix {prefix} length {prefix.Length} is too long");
    return string.Create(availableLength, 0, (span, _) =>
    {
        for (int i = 0; i < span.Length; i++)
            span[i] = (char)('a' + Random.Shared.Next(0, 26));
    });
}
```

Reply:
Stephen thank you for the clarification!

Reply:
I ran this code locally on Android and the respective tests passed.